src/main/interface/cli.c

```diff
@@ -908,6 +908,46 @@ static void cliSerial(char *cmdline)
 }
 
 #ifndef SKIP_SERIAL_PASSTHROUGH
+static IO_t serialPassthroughDtrPin = IO_NONE;
+
+// Callback routine registered with CLI serial port to support Arduino programming via passthrough
+//
+// For example to program a MinimOSD connected to UART 6 on an F4 with port C8 used
+// for DTR (eg OpenPilot Revolution receiver port)
+//
+// Use 'serialpassthrough 5 57600 rxtx c8' and then use Ardino to program MinimOSD
+// Use 'serialpassthrough 5 115200' and then use MWOSD configurator to setup MinimOSD
+//
+static void cbCtrlLine(uint16_t data)
```

Comment:
`cli.c` isn't really the place for this. How about `drivers/serial.c`?

Reply:
The pin is only used in cli, it's not serial related ...

Reply:
To my understanding it is used for serial passthrough, which is initiated in CLI, but I would not consider it to be part of CLI. Maybe it needs a new file, like the ESC passthrough...

Reply:
`serialPassthrough()` is in io/serial.c ..
moving cbCtrlLine + serialPassthroughDtrPin will either pollute global namespace or will need interface functions ...

Reply:
Could / should `serialPassthrough()` take care of handling the DTR pin, and should the config be added to this call?

Reply:
Not sure ... the code is simpler with CLI handling it ...

Reply:
Potentially simpler, but not nicely structured. In terms of maintenance, having all of the serial passthrough code in one place will be easier.

Reply:
I've taken the time to look at the serial driver structure and it doesn't really lend itself to the addition of DTR handling for a couple reasons. Because the openSerialPort() call takes parameters taken from the serialPortConfig_t structure rather than simply a pointer to the structure, expansion of it's parameters to include a DTR pin requires modification of every call to openSerialPort() even if most drivers won't use DTR. I think it simpler to consider the DTR pin usage to be a sideband signal, not really part of the serial port, but rather a reset line controlled by USB.

Reply:
It should not go into the `driver` layer, but into `io`, where `serialPassthrough()` already is.